src/client/entity-editor/series-section/series-section.tsx

```diff
@@ -0,0 +1,159 @@
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+import * as Immutable from 'immutable';
+import * as React from 'react';
+import {Action, updateOrderType, updateSeriesType} from './actions';
+import {Col, Row} from 'react-bootstrap';
+
+import CustomInput from '../../input';
+import type {Dispatch} from 'redux';
+import Select from 'react-select';
+import {connect} from 'react-redux';
+import {sortRelationships} from '../relationship-editor/actions';
+
+
+type SeriesOrderingType = {
+	label: string,
+	id: number
+};
+
+type StateProps = {
+	orderTypeValue: number,
+	seriesTypeValue: string,
+	relationships: Immutable.List<any>
+};
+
+
+type DispatchProps = {
+	onOrderTypeChange: (obj: {value: number}) => unknown,
+	onSeriesTypeChange: (obj: {value: string}) => unknown
+};
+
+type OwnProps = {
+	seriesOrderingTypes: Array<SeriesOrderingType>
+};
+
+type Props = StateProps & DispatchProps & OwnProps;
+
+/**
+ * Container component. The SeriesSection component contains input fields
+ * specific to the series entity. The intention is that this component is
+ * rendered as a modular section within the entity editor.
+ *
+ * @param {Object} props - The properties passed to the component.
+ * @param {Array} props.seriesOrderingTypes - The list of possible ordering
+ * 		  types for a series.
+ * @param {number} props.orderTypeValue - The ID of the ordering type currently selected for
+ *        the series.
+ * @param {string} props.seriesTypeValue - The value of the entity type currently selected for
+ *        the series.
+ * @param {Immutable.List<any>} props.relationships - The list of relationships conatined by
+ * 		  the series.
+ * @param {Function} props.onOrderTypeChange - A function to be called when
+ *        a different ordering type is selected.
+ * @param {Function} props.onSeriesTypeChange - A function to be called when
+ *        a different series type is selected.
+ * @returns {ReactElement} React element containing the rendered
+ *        SeriesSection.
+ */
+function SeriesSection({
+	seriesOrderingTypes,
+	orderTypeValue,
+	seriesTypeValue,
+	relationships,
+	onOrderTypeChange,
+	onSeriesTypeChange
+}: Props) {
+	const relationshipsObject = relationships.toJS();
+	const relationshipsArray = Object.values(relationshipsObject);
+	const seriesOrderingTypesForDisplay = seriesOrderingTypes.map((type) => ({
+		label: type.label,
+		value: type.id
+	}));
```

Comment:
All this we should do in mapStateToProps instead

Reply:
Line 86-89 is fine here 🤔

Reply:
Better to do it in mapStateToProps so that it only get recalculated if the state changes, rather than on every re-render of the component( which might rerender for reasons other than a state change)

Reply:
Although, let's be honest, for the two items in this array it won't make a difference :D